### PR TITLE
Fix to #724

### LINKED ISSUE #724: B Lin Regression: NaN's without explanation
the NaN's in the following screenshot are because one of the independents contain infinity. we always try and explain why a value can't be calculated.

screen shot 2015-06-18 at 8 46 27 pm

cheers

### PR DESCRIPTION
**Pull request by mmarsman1**
This fixes #724 and affects:
 — Bayesian ANCOVA
 — Bayesian Linear Regression
 — Bayesian RM ANOVA

### CODE REVIEW

File: JASP-Engine/JASP/R/commonbayesianlinearmodels.R [hunk 1]
```diff
@@ -86,6 +86,16 @@
 		error.message <- "Bayes factor is undefined -- the dependent variable contains infinity"
 
 	if (!exists ("error.message") && perform == "run") {
+		covariate.names <- NULL
+		for (covariate in options$covariates) {
+			if (any (!is.finite (dataset [[.v (covariate)]])))
+				covariate.names <- c (covariate.names, covariate)
+		}
+		if ( !is.null (covariate.names))
+			error.message <- paste ("Bayes factor is undefined -- the dependent variable(s) ", covariate.names, " contain(s) infinity", sep = "")
```
Comment (jonathon-love) [inline JASP-Engine/JASP/R/commonbayesianlinearmodels.R]: could this be generalised: "the variable(s) ... contains"

as i expect an infinity in the dependent or independents would be problematic.

Comment (mmarsman1) [inline JASP-Engine/JASP/R/commonbayesianlinearmodels.R]: No

On 18 Jun 2015, at 21:33, Jonathon Love <notifications@github.com<mailto:notifications@github.com>> wrote:


In JASP-Engine/JASP/R/commonbayesianlinearmodels.R<https://github.com/jasp-stats/jasp-desktop/pull/725#discussion_r32769172>:

> @@ -86,6 +86,16 @@
>               error.message <- "Bayes factor is undefined -- the dependent variable contains infinity"
>
>       if (!exists ("error.message") && perform == "run") {
> +             covariate.names <- NULL
> +             for (covariate in options$covariates) {
> +                     if (any (!is.finite (dataset [[.v (covariate)]])))
> +                             covariate.names <- c (covariate.names, covariate)
> +             }
> +             if ( !is.null (covariate.names))
> +                     error.message <- paste ("Bayes factor is undefined -- the dependent variable(s) ", covariate.names, " contain(s) infinity", sep = "")


could this be generalised: "the variable(s) ... contains"

as i expect an infinity in the dependent or independents would be problematic.

Comment (jonathon-love) [inline JASP-Engine/JASP/R/commonbayesianlinearmodels.R]: oic, you handle the dependent above. then shouldn't this read "the independent variable contains infinity?"

**Comment by mmarsman1** [comment]
Yes

**Pull request merged**